tools/utilities/CheckpointToCSV/CheckpointFileUtil.H

```diff
@@ -0,0 +1,107 @@
+#ifndef CHECKPOINTFILEUTIL_H_
+#define CHECKPOINTFILEUTIL_H_
+#include <AMReX_Config.H>
+
+#include <AMReX_Geometry.H>
+#include <AMReX_MultiFab.H>
+#include <AMReX_VisMF.H>
+#include <string>
+#include <memory>
+
+using namespace amrex;
+
+class CheckpointFileDataImpl
+{
+public:
+    CheckpointFileDataImpl(
+        std::string const& chkptfile_name,
+        Vector<std::string> addl_field_names = {});
+
+    [[nodiscard]] int spaceDim() const noexcept { return m_spacedim; }
+
+    [[nodiscard]] Real time() const noexcept { return m_time; }
+
+    [[nodiscard]] int finestLevel() const noexcept { return m_finest_level; }
+
+    [[nodiscard]] const BoxArray& boxArray(int level) const noexcept
+    {
+        return m_ba[level];
+    }
+
+    [[nodiscard]] const DistributionMapping&
+    DistributionMap(int level) const noexcept
```

Comment:
totally minor: you seem to be following camelCase for all the other member functions but this one.

Reply:
true! this is straight from "amrex/Src/Base/AMReX_PlotFileDataImpl.H" though